instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanSuppressors.java

```diff
@@ -63,24 +63,24 @@ public boolean shouldSuppress(Context parentContext, SpanKind spanKind) {
 
   static final class BySpanKey implements SpanSuppressor {
 
-    private final Set<SpanKey> spanKeys;
+    private final SpanKey[] spanKeys;
 
     BySpanKey(Set<SpanKey> spanKeys) {
-      this.spanKeys = spanKeys;
+      this.spanKeys = spanKeys.toArray(new SpanKey[spanKeys.size()]);
     }
 
     @Override
     public Context storeInContext(Context context, SpanKind spanKind, Span span) {
-      for (SpanKey spanKey : spanKeys) {
-        context = spanKey.storeInContext(context, span);
+      for (int i = 0; i < spanKeys.length; i++) {
+        context = spanKeys[i].storeInContext(context, span);
       }
       return context;
     }
 
     @Override
     public boolean shouldSuppress(Context parentContext, SpanKind spanKind) {
-      for (SpanKey spanKey : spanKeys) {
-        if (spanKey.fromContextOrNull(parentContext) == null) {
+      for (int i = 0; i < spanKeys.length; i++) {
```

Comment:
for arrays it should be ok to use the extended for syntax, these loops should be compiled similarly

Reply:
Right you are - the bytecode doesn't allocate anything (several `javac` versions) and the runtime is as expected (starts fast, stays fast).  Fixed.